### PR TITLE
add multi-level verbosity to the logging

### PR DESCRIPTION
**Pull request by bradzacher**
I wanted a way to silence the all but the most important logs, to reduce a bit of the clutter in my command window.

Verbose is now a Number option:
- `0` = warnings
- `1` = warnings + info
- `2` = warnings + info + verbose

Has handling for converting the old boolean property value to the Number value:
- `false` maps to `1`
- `true` maps to `2`

### CODE REVIEW

File: lib/server.js [hunk 1]
```diff
@@ -16,6 +16,12 @@ var ProcessManager = function () {
 ProcessManager.prototype = new EventEmitter();
 ProcessManager.prototype.init = function (opt) {
   this.opt = opt;
+  // handle old boolean verbose values
+  if (this.opt.verbose === true) {
+    this.opt.verbose = 2;
+  } else if (this.opt.verbose === false) {
+    this.opt.verbose = 1;
+  }
```
Comment (ssreekanth) [inline lib/server.js]: Same comments as I made for the code changes in lib/client.js apply here too.

File: lib/client.js [hunk 2]
```diff
@@ -51,6 +51,12 @@ Client.prototype.join = function (browserWindow, options, cb) {
     }
     browserWindow = getBrowserWindow();
   }
+  // handle old boolean verbose values
+  if (this.opt.verbose === true) {
+    this.opt.verbose = 2;
+  } else if (this.opt.verbose === false) {
+    this.opt.verbose = 1;
+  }
```
Comment (ssreekanth) [inline lib/client.js]: Change `this.opt.verbose` to `this.opt.logLevel`. And, use constant names instead of numbers.

File: lib/client.js [hunk 3]
```diff
@@ -15,7 +15,7 @@ var Client = function () {};
 var defaultOpt = {
   port: 30080,
   sendBounds: true,
-  verbose: false
+  verbose: 1
```
Comment (bradzacher) [inline lib/client.js]: would you like to continue keep backward compatibility with verbose like I had already implemented?

or just remove verbose all together?

Comment (ssreekanth) [inline lib/client.js]: Good point. Lets keep backward compatibility with `verbose`, but transform`verbose` to `logLevel` internally. However, make changes to the documentation in `README.md` by replacing `verbose` to `logLevel`.

File: README.md [hunk 4]
```diff
@@ -95,7 +95,7 @@ If you want details, see [example/simple](example/simple).
  * `path` String. A path to your `package.json` file (default: `process.cwd()`).
  * `port` Number. WebSocket server port (default: `30080`).
  * `spawnOpt` Object. Options for [spawn](https://nodejs.org/api/child_process.html#child_process_child_process_spawn_command_args_options).
- * `verbose` Boolean. If set, show all electron-connect log in your prompt (default: `false`).
+ * `verbose` Number. The granularity of the electron-connect logging in your prompt. `0` - warning only, `1` - warning and info only, `2` - all logs (default: `1`).
```
Comment (ssreekanth) [inline README.md]: As mentioned in the other comment, replace `verbose` with `logLevel` here.

File: README.md [hunk 5]
```diff
@@ -167,7 +167,7 @@ Broadcasts an event to all clients.
 * `options` Object
  * `port` Number. WebSocket server port (default: `30080`) that client should connect to.
  * `sendBounds` Boolean
- * `verbose` Boolean
+  * `verbose` Number. `0` - warning only, `1` - warning and info only, `2` - all logs (default: `1`).
```
Comment (ssreekanth) [inline README.md]: As mentioned in the other comment, replace `verbose` with `logLevel` here.

**Comment by bradzacher** [comment]
@ssreekanth changes made and tested okay

File: lib/client.js [hunk 6]
```diff
@@ -51,6 +51,14 @@ Client.prototype.join = function (browserWindow, options, cb) {
     }
     browserWindow = getBrowserWindow();
   }
+
+  // handle old boolean verbose values
+  if (this.opt.verbose === true) {
+    this.opt.verbose = util.LOG_LEVEL_VERBOSE;
```
Comment (ssreekanth) [inline lib/client.js]: it should be `this.opt.logLevel = ....` here

File: lib/client.js [hunk 7]
```diff
@@ -51,6 +51,14 @@ Client.prototype.join = function (browserWindow, options, cb) {
...
+  } else if (this.opt.verbose === false) {
+    this.opt.verbose = util.LOG_LEVEL_INFO;
```
Comment (ssreekanth) [inline lib/client.js]: it should be `this.opt.logLevel = ....`

File: lib/server.js [hunk 8]
```diff
@@ -16,6 +17,14 @@ var ProcessManager = function () {
 ProcessManager.prototype = new EventEmitter();
 ProcessManager.prototype.init = function (opt) {
   this.opt = opt;
+  
+  // handle old boolean verbose values
+  if (this.opt.verbose === true) {
+    this.opt.verbose = util.LOG_LEVEL_VERBOSE;
```
Comment (ssreekanth) [inline lib/server.js]: it should be `this.opt.logLevel = ....`

File: lib/server.js [hunk 9]
```diff
@@ -16,6 +17,14 @@ var ProcessManager = function () {
...
+  } else if (this.opt.verbose === false) {
+    this.opt.verbose = util.LOG_LEVEL_INFO;
```
Comment (ssreekanth) [inline lib/server.js]: it should be `this.opt.logLevel = ....`

**Comment by bradzacher** [comment]
@ssreekanth - sorry, i cooked a bad batch of copy pasta

**Comment by ssreekanth** [comment]
@bradzacher Merged. Thank you! 👍

**Pull request merged**